### PR TITLE
Issue #13328: Kill mutation for PackageNamesLoader

### PR DESCRIPTION
**Pull request by Kevin222004**
Issue #13328: Kill mutation for PackageNamesLoader

# Mutation 
https://github.com/checkstyle/checkstyle/blob/master/config/pitest-suppressions/pitest-packagenamesloader-suppressions.xml

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/PackageNamesLoader.java [hunk 1]
```diff
@@ -187,7 +187,7 @@ private static void processFile(URL packageFile, PackageNamesLoader namesLoader)
      *
      * @return map between local resources and dtd ids.
      */
-    private static Map<String, String> createIdToResourceNameMap() {
+    public static Map<String, String> createIdToResourceNameMap() {
```
Comment (Kevin222004) [inline src/main/java/com/puppycrawl/tools/checkstyle/PackageNamesLoader.java:190]: This case looks not possible to cover. But to cover the test it is required to make it public.

Please give your suggestion if it is not fine

**Comment by Kevin222004** [comment]
Failures are not related to changes

**Review by nrmancuso — CHANGES_REQUESTED** [review]
@Kevin222004 please try:

```
    private PackageNamesLoader()
            throws ParserConfigurationException, SAXException {
        super(Map.of(
            DTD_PUBLIC_ID, DTD_RESOURCE_NAME,
            DTD_PUBLIC_CS_ID, DTD_RESOURCE_NAME
        ));
    }
```

And remove method completely.

**Comment by Kevin222004** [comment]
> @Kevin222004 please try:
> 
> ```
>     private PackageNamesLoader()
>             throws ParserConfigurationException, SAXException {
>         super(Map.of(
>             DTD_PUBLIC_ID, DTD_RESOURCE_NAME,
>             DTD_PUBLIC_CS_ID, DTD_RESOURCE_NAME
>         ));
>     }
> ```
> 
> And remove the method completely.

Done, Thanks :), I don't get this idea in my mind while working on this pr

**Comment by Kevin222004** [comment]
@nrmancuso @romani @stoyanK7 I don't know you are aware of this or not but 
```
[INFO]      [echo] Checkstyle started (checkstyle-input-checks.xml): 23/06/2023 10:16:12 am
[INFO] [checkstyle] Running Checkstyle  on 2109 files
[INFO]      [echo] Checkstyle finished (checkstyle-input-checks.xml): 23/06/2023 10:17:22 am
[INFO]      [echo] Checkstyle started (checkstyle-examples-checks.xml): 23/06/2023 10:16:12 am
[INFO] [checkstyle] Running Checkstyle  on 54 files
[ERROR] [checkstyle] [ERROR] /home/kevin/Desktop/check_style/checkstyle/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/singlespaceseparator/Example1.java:14: Trailing whitespace is not allowed [noTrailingWhitespace]
[ERROR] [checkstyle] [ERROR] /home/kevin/Desktop/check_style/checkstyle/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/singlespaceseparator/Example1.java:15: Trailing whitespace is not allowed [noTrailingWhitespace]
[ERROR] [checkstyle] [ERROR] /home/kevin/Desktop/check_style/checkstyle/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/singlespaceseparator/Example2.java:25: Trailing whitespace is not allowed [noTrailingWhitespace]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:48 min
[INFO] Finished at: 2023-06-23T10:18:46+05:30

```

This error is from master branch might be from https://github.com/checkstyle/checkstyle/pull/13272

**Comment by stoyanK7** [comment]
@Kevin222004 Thanks for the tip. Kind of expected because we didn't have the checks when these examples got merged. I opened PR https://github.com/checkstyle/checkstyle/pull/13294. Please disregard these errors.

**Review by nrmancuso — APPROVED** [review]
PR is rebased, ok to merge.

**Comment by Kevin222004** [comment]
@rdiachenko ping

**Review by rdiachenko — APPROVED** [review]
lgtm

File: src/main/java/com/puppycrawl/tools/checkstyle/PackageNamesLoader.java [hunk 2]
```diff
@@ -84,7 +83,11 @@ public final class PackageNamesLoader
      */
     private PackageNamesLoader()
             throws ParserConfigurationException, SAXException {
-        super(createIdToResourceNameMap());
+        // map between local resources and dtd ids
+        super(Map.of(
+            DTD_PUBLIC_ID, DTD_RESOURCE_NAME,
+            DTD_PUBLIC_CS_ID, DTD_RESOURCE_NAME
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/PackageNamesLoader.java:89]: same as https://github.com/checkstyle/checkstyle/pull/13299#discussion_r1243693995

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/main/java/com/puppycrawl/tools/checkstyle/PackageNamesLoader.java (hunk 2 above)
Comment (Kevin222004) [reply src/main/java/com/puppycrawl/tools/checkstyle/PackageNamesLoader.java]: Done Please check Test added

**Comment by Kevin222004** [comment]
@romani only test added still linkcheck failing

**Review by romani — APPROVED** [review]
Ok to merge 

Just tests were added

**Pull request merged**